logging/telemetryConfig_test.go

```diff
@@ -90,3 +90,23 @@ func Test_CreateSaveLoadTelemetryConfig(t *testing.T) {
 	a.Equal(config1.Password, config2.Password)
 
 }
+
+func Test_SanitizeTelemetryString(t *testing.T) {
+	type testcase struct {
+		input    string
+		expected string
+		parts    int
+	}
+
+	tests := []testcase{
+		{"no:change", "no:change", 2},
+		{"@@@@@@@remove(-some*&!-garbage", "remove-some-garbage", 1},
+		{"@@@@@@@remove(-some*&!-garbage:times-%two", "remove-some-garbage:times-two", 2},
+		{"this.needs.to.be.truncated.because.it.is.way.too.long", "this.needs.to.be.truncated.because.it.i", 1},
+		{"2001:0db8:85a3:0000:0000:8a2e:0370:7334", "2001:0db8:85a3:0000:0000:8a2e:0370:7334", 1},
```

Comment:
this works as it passes through strings.Join(["2001","0db8:85a3:0000:0000:8a2e:0370:7334"], ":")
if the intention was to split("{IPv6 addr}:{something else}") and make sure neither side was too long, it isn't that.
Does this even matter? Is the left hand side likely to be an IPv6 addr?
e.g. this test case should fail:
`{"2001:0db8:85a3:0000:0000:8a2e:0370:7334:whaarrrrrrgarrrrbleeeeeeeeeefoobarbazbearquxzapzowyow", "2001:0db8:85a3:0000:0000:8a2e:0370:7334:whaarrrrrrgarrrrbleeeeeeeeeefoobarbazbe", 1},`

Reply:
I think it matters because this is where I'm using the sanitizer:
```diff
func getCommonHeaders(headers http.Header) (otherTelemetryGUID, otherInstanceName, otherPublicAddr string) {
- 	otherTelemetryGUID = headers.Get(TelemetryIDHeader)
- 	otherInstanceName = headers.Get(InstanceNameHeader)
- 	otherPublicAddr = headers.Get(AddressHeader)
+ 	otherTelemetryGUID = logging.SanitizeTelemetryString(headers.Get(TelemetryIDHeader), 1)
+ 	otherInstanceName = logging.SanitizeTelemetryString(headers.Get(InstanceNameHeader), 2)
+ 	otherPublicAddr = logging.SanitizeTelemetryString(headers.Get(AddressHeader), 1)
	return
}
```

You're right that it is a little curious that these two tests pass: 
`{"2001:0db8:85a3:0000:0000:8a2e:0370:7334", "2001:0db8:85a3:0000:0000:8a2e:0370:7334", 1}`
and
`{"2001:0db8:85a3:0000:0000:8a2e:0370:7334", "2001:0db8:85a3:0000:0000:8a2e:0370:7334", 8}`

One thing to note is that in the case where max parts is 1, the result is an array whose single element is the input. With these tests you can validate that the ipv6 address truncates anything > 39 characters, i.e. this test passes:
`{"2001:0db8:85a3:0000:0000:8a2e:0370:7334:9999", "2001:0db8:85a3:0000:0000:8a2e:0370:7334", 1}`

Reply:
This is all N/A now. No longer trying to parse the inputs, just truncate them if they're big.